drafts/25-xxx-specifications.txt

```diff
@@ -567,11 +567,122 @@ nd20. The result of evaluating a processor-dependent directive is processor-depe
+      as divide by zero).
+
+ex25. When the expression evaluates to zero, the controlling expression
+      will be considered "false". If the expression evaluates to
+      any non-zero value, the controlling expression will be considered
+      "true".
+
+
+5.1 Primary expressions
+-----------------------
+
+Preprocessing recognizes the following primary expressions in
+controlling expressions.
+
+Since expression evaluation occurs *after* token expansion, there will
+be no object-like macros or function-like macros left to evaluate. All
+instances of ID or ID (args) will all have been replaced with their
+expansions.
+
+We list them here for completeness in describing the illustrative
+syntax of controlling expressions.
+
+| Primary      | Evaluation semantics                        |
+|--------------+---------------------------------------------|
+| ID           | The expansion of the object-like macro ID   |
+| ID (args)    | The expansion of the function-like macro ID |
+| WHOLE_NUMBER | Decimal value of WHOLE_NUMBER               |
+| ( expr )     | Parenthesized expressions                   |
+|--------------+---------------------------------------------|
+
+
+
+5.1 Operators allowed in controlling expressions
+------------------------------------------------
+
+To maintain compatibility with the use of C preprocessing directives
+in many existing Fortran programs, the operators allowed in
+controlling expressions in #if and #elif expressions are a subset of
+those defined in the C 2023 standard §6.5 "Expressions" and §6.6
+"Constant expressions".
+
+A "precedence level" is assigned to each operator that determines
+how the operators combine with sub-expressions containing other operators
+at different precedence levels.
+
+An "associativity" is assigned to each operator that determines how
+operators at the same precedence level are combined.
+    - "left" means the operator binds to the left,
+    - "right" means the operator binds to the right.
+    - "nonassoc" means that the operator is not associative.
+
+The following table describes the semantics of the allowed operators
+in conditional expressions. The table is grouped by precedence level,
+from lowest precedence to highest.
+
+We label subexpressions "e1", "e2", and "e3" to aid in describing the
+evaluation semantics. Unless otherwise specified, all operators
+evaluate with the same semantics as their Fortran counterparts.
+
+
+| Prec |   Op    |    Syntax    |  Assoc'y |  Evaluation Semantics      |
+|------+---------+--------------+----------+----------------------------|
+| low  |   ? :   | e1 ? e2 : e3 |  right   |      conditional-expr      |
+|------+---------+--------------+----------+----------------------------|
+|      |   ¦¦    |   e1 || e2   |   left   |       Fortran .OR.         |
+|------+---------+--------------+----------+----------------------------|
+|      |   &&    |   e1 && e2   |   left   |       Fortran .AND.        |
```

Comment:
In the 5/12 call we resolved these should be short-circuit evaluation as in CPP, to allow things like:
```
#if  x && 1/x
#endif
```
which means they are NOT simply  Fortran .OR. / .AND.